crates/provider/src/native.rs

```
@@ -0,0 +1,601 @@
+                args.remove(0);
+            }
+        }
+
+        let output = Command::new("sh")
+            .arg("-c")
+            .arg(args.join(" "))
+            .output()
+            .await?;
+
+        if !output.stdout.is_empty() {
+            return Ok(RunCommandResponse {
+                exit_code: output.status,
+                std_out:   output.stdout,
+                std_err:   None,
+            });
+        } else if !output.stderr.is_empty() {
+            if !opts.allow_fail {
+                return Err(Box::new(std::io::Error::new(
+                    ErrorKind::Other,
+                    "Allow fail",
+                )));
+            };
+
+            return Ok(RunCommandResponse {
+                exit_code: output.status,
+                std_out:   output.stdout,
+                std_err:   Some(output.stderr),
+            });
+        }
+
+        Ok(RunCommandResponse {
+            exit_code: output.status,
+            std_out:   output.stdout,
+            std_err:   Some(output.stderr),
+        })
+    }
+
+    async fn create_resource(&mut self, resourse_def: PodDef) -> Result<(), Box<dyn Error>> {
+        let name: String = resourse_def.metadata.name.clone();
+
+        // This is temporary solution for filling up the process map. To be deleted:
+        self.process_map.insert(
+            name.clone(),
+            Process {
+                pid:          1,
+                log_dir:      format!("{}/{}", self.tmp_dir, name.clone()),
+                port_mapping: HashMap::new(),
+                command:      String::new(),
+            },
+        );
+        // Delete the code above once spawnFromDef is implemented
+
+        let local_file_path: String = format!("{}/{}.yaml", &self.tmp_dir, name);
+        let content: String = serde_json::to_string(&resourse_def)?;
+
+        self.filesystem
+            .write(&local_file_path, content)
+            .expect("Create source: Failed to write file");
```

We should return the Error.